### PR TITLE
Update light.py

### PR DESCRIPTION
**Pull request by bluedragon456**
setup to auto-load the new water timer. I haven't figured out the outlet plug it comes with that acts as the Wi-Fi hub since the timer is only Bluetooth-based.

### CODE REVIEW

**Comment by darrenblanchard** [comment]
Not sure this will work I have a hubspace water timer and did the original change to code, unless the API change recently, the model shows up as TBD, this was why it used the device class to identify and create the entity, is your showing a Model ?

Darren

**Comment by darrenblanchard** [comment]
The plug showed up for me without any changes.

**Comment by bluedragon456** [comment]
From what I had this wouldn’t auto load I could add manually entering the friendlyname but it would stop my other devices from being seen. Unless I added them as well. Which was a problem due to it being a ceiling fan with light. Auto use the childnames and the friendlyname is the parent name which has issue controlling the fan from what I was experiencing.

**Comment by darrenblanchard** [comment]
I use friendly names, did you get a model? because I did not it shows as TBD, given that I suggest using an "or" in the if statement. I will check my firmware version.

Darren

On Jun 1, 2024 1:05 PM, bluedragon456 ***@***.***> wrote:

From what I had this wouldn’t auto load I could add manually entering the friendlyname but it would stop my other devices from being seen. Unless I added them as well. Which was a problem due to it being a ceiling fan with light. Auto use the childnames and the friendlyname is the parent name which has issue controlling the fan from what I was experiencing.

**Comment by darrenblanchard** [comment]
my firmware versions are 2.5.25670-water and 1.0.20 I assume they are latest it's not asking me to upgrade.

On Jun 1, 2024 1:05 PM, bluedragon456 ***@***.***> wrote:

From what I had this wouldn’t auto load I could add manually entering the friendlyname but it would stop my other devices from being seen. Unless I added them as well. Which was a problem due to it being a ceiling fan with light. Auto use the childnames and the friendlyname is the parent name which has issue controlling the fan from what I was experiencing.

**Comment by bluedragon456** [comment]
It doesn't show the model. Just TBD. Checking the firmware it matches what you are getting.

**Comment by darrenblanchard** [comment]
Your code used model, that was the reason to code using the device class instead.

I have not had time to code the battery life.

Darren

On Jun 1, 2024 4:16 PM, bluedragon456 ***@***.***> wrote:

It doesn't show the model. Just TBD. Checking the firmware it matches what you are getting.

**Comment by bluedragon456** [comment]
Ah I see what your talking about that was one of the things I tried. There is an issue log for the water timer and it said that it maybe better to pull the model so I tried that but for what I was going for it made no difference. Also seeing a few other spots I changed that I hadn't realized didn't get changed back.

File: custom_components/hubspace/light.py [hunk 1]
```diff
@@ -95,7 +95,7 @@ def _add_entity(entities, hs, model, deviceClass, friendlyName, debug):
         entities.append(HubspaceFan(hs, friendlyName, debug))
         _LOGGER.debug("Creating Light")
         entities.append(HubspaceLight(hs, friendlyName, debug))
-    elif model == "DriskolFan" or model == "ZandraFan" or model == "TagerFan" or model == "VinwoodFan" or model == "CF2003":
```
Comment (bluedragon456) [inline custom_components/hubspace/light.py:98]: Not sure why the one I have removed this model it wasn't something I did.

**Comment by bluedragon456** [comment]
I went back through and removed the changes I had made that were of no help and update the PR

**Comment by jdeath** [comment]
Is this PR ready to go? Many models show up at TBD as they choose not to fill it in. That is why I sometimes need to check on the default icon or some other field to determine the model number...

My home assistant installation is messed up, so I cannot test right now

**Comment by bluedragon456** [comment]
Pulled the latest code and added my code. So it should be good to go.

**Pull request merged**